src/core/i18n/translate.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component, PropTypes } from 'react';
+
+
+function getDisplayName(component) {
+  return component.displayName || component.name || 'Component';
+}
+
+
+export default function translate(options = {}) {
+  const { withRef = false } = options;
+
+  return function Wrapper(WrappedComponent) {
+    class Translate extends Component {
+      constructor(props, context) {
+        super(props, context);
+        this.i18n = context.i18n;
+        this.state = {
+          i18nLoadedAt: null,
+        };
+      }
+
+      getChildContext() {
+        return { i18n: this.i18n };
+      }
+
+      getWrappedInstance() {
+        if (!withRef) {
+          throw new Error(dedent`To access the wrapped instance, you need to specify
+            { withRef: true } as the second argument of the translate() call.`);
+        }
+        return this.refs.wrappedInstance;
+      }
+
+      render() {
+        const extraProps = { i18n: this.i18n };
+
+        if (withRef) {
+          extraProps.ref = 'wrappedInstance';
+        }
+
+        return React.createElement(
```

Comment:
This should probably be `return <WrappedComponent {...this.props, ...extraProps} />`